### PR TITLE
Caulaflower apparel auto patcher

### PR DESCRIPTION
**Pull request by Caulaflower**
## Additions

Added autopatcher for apparel

## Changes

Changed quality statpart to one which can be disabled with a mod extension

## Reasoning

Autopatcher is pretty self explanatory

Boldi complained about the quality scaling

## Alternatives
Do these sections matter tbh

## Testing

Check tests you have performed:
- [x] Compiles without warnings
- [x] Game runs without errors

### CODE REVIEW

File: Source/CombatExtended/CombatExtended/StatParts/StatPart_QualityConditional.cs [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+		{
+			if (req.Def.GetModExtension<QualityConditionalModExt>()?.UseQualityMult ?? true)
+			{
+				if (!(val <= 0f) || applyToNegativeValues)
+				{
+					float a = val * QualityMultiplier(req.QualityCategory) - val;
+					a = Mathf.Min(a, MaxGain(req.QualityCategory));
+					val += a;
+				}
+			}
```
Comment (MadaraUchiha) [inline Source/CombatExtended/CombatExtended/StatParts/StatPart_QualityConditional.cs:54]: Prefer early returns to nested `if` statements

```suggestion
			if (!req.Def.GetModExtension<QualityConditionalModExt>()?.UseQualityMult ?? false)
			{
				return;
			}
			if (val <= 0f && !applyToNegativeValues)
			{
				return;
			}
			float a = val * QualityMultiplier(req.QualityCategory) - val;
			a = Mathf.Min(a, MaxGain(req.QualityCategory));
			val += a;
```

File: Source/CombatExtended/CombatExtended/StatParts/StatPart_QualityConditional.cs [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+		}
+
+		public override string ExplanationPart(StatRequest req)
```
Comment (MadaraUchiha) [inline Source/CombatExtended/CombatExtended/StatParts/StatPart_QualityConditional.cs:58]: Similarly in this method, you've already started w/ early returns, keep at it.

The opposite of `if (optional?.shit() ?? true)` is `if (!optional?.shit() ?? false)`

File: Source/CombatExtended/Compatibility/ApparelAutoPatcher.cs [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+                       new StatModifier { value = preset.FinalRatingBlunt(apparel.GetStatValueDef(StatDefOf.StuffEffectMultiplierArmor)), stat = StatDefOf.ArmorRating_Blunt }
+                       };
+
+                    }
+
+                   
+
+                    apparel.statBases.RemoveAll(x => 
+                    x.stat == StatDefOf.ArmorRating_Sharp 
+                    |
```
Comment (MadaraUchiha) [inline Source/CombatExtended/Compatibility/ApparelAutoPatcher.cs:55]: `||`

**Review by MadaraUchiha — DISMISSED** [review]
LGTM - Similarly to the previous one, nothing too groundbreaking, take or ignore comments as you wish :)

**Review by N7Huntsman — APPROVED** [review]
Seems to work, though some of the stuffable armor material thickness seems low in some cases.

**Pull request merged**